mining.go

```diff
@@ -1105,24 +1098,15 @@ func NewBlockTemplate(policy *mining.Policy, server *server, payToAddress dcruti
 	// 8. Select the one with the largest penalty ratio (highest block reward).
 	//    This block is then selected to build upon instead of the others, because
 	//    it yields the greater amount of rewards.
-	chainState.Lock()
-	prevHash := chainState.newestHash
-	nextBlockHeight := chainState.newestHeight + 1
-	poolSize := chainState.nextPoolSize
-	reqStakeDifficulty := chainState.nextStakeDifficulty
-	finalState := chainState.nextFinalState
-	winningTickets := make([]chainhash.Hash, len(chainState.winningTickets))
-	copy(winningTickets, chainState.winningTickets)
-	missedTickets := make([]chainhash.Hash, len(chainState.missedTickets))
-	copy(missedTickets, chainState.missedTickets)
-	chainState.Unlock()
-
-	chainBest := blockManager.chain.BestSnapshot()
-	if *prevHash != chainBest.Hash ||
-		nextBlockHeight-1 != chainBest.Height {
+
+	best := blockManager.chain.BestSnapshot()
+	prevHash := best.Hash
+	nextBlockHeight := best.Height + 1
+	if prevHash != best.Hash ||
```

Comment:
lol
https://github.com/decred/dcrd/pull/1419#discussion_r216648859

@dnldd you should probably add a comment or modify the first comment of the PRs so that their sequencing is explicit (I didn't realize this one was coming before #1418, though looking back now I should have looked at the PRs in the reverse order github presents them, so really _mea culpa_).

Reply:
Agreed.  I generally preface every PR in a series with **This PR requires #xxxxx**.

Reply:
Noted, will get that done.